$NetBSD$

# clean up the FreeBSD configuration to fit better with pkgsrc

--- build/config/FreeBSD.orig	2010-01-15 12:13:34.000000000 +0000
+++ build/config/FreeBSD
@@ -20,7 +20,8 @@ CXX     = g++
 LINK    = $(CXX)
 LIB     = ar -cr
 RANLIB  = ranlib
-SHLIB   = $(CXX) -shared -Wl,-soname,$(notdir $@) -o $@
+SHLIBFLAGS = -L$(POCO_PREFIX)/lib -Wl,-R$(POCO_PREFIX)/lib
+SHLIB   = $(CXX) -Wl,-R$(POCO_PREFIX)/lib -shared -Wl,-soname,$(notdir $@) -o $@
 SHLIBLN = $(POCO_BASE)/build/script/shlibln
 DEP     = $(POCO_BASE)/build/script/makedepend.gcc 
 STRIP   = strip
@@ -52,7 +53,7 @@ STATICOPT_CXX   =
 STATICOPT_LINK  = -static
 SHAREDOPT_CC    = -fPIC
 SHAREDOPT_CXX   = -fPIC
-SHAREDOPT_LINK  = -Wl,-rpath,$(LIBPATH)
+SHAREDOPT_LINK  = -Wl,-rpath,$(POCO_PREFIX)/lib -L$(LIBPATH)
 DEBUGOPT_CC     = -g -D_DEBUG
 DEBUGOPT_CXX    = -g -D_DEBUG
 DEBUGOPT_LINK   = -g
